code/datums/elements/light_eater.dm

```diff
@@ -0,0 +1,171 @@
+/**
+ * Makes anything it attaches to capable of permanently removing something's ability to produce light.
+ *
+ * The temporary equivalent is [/datum/component/light_eater]
+ */
+/datum/element/light_eater
+	element_flags = ELEMENT_DETACH
+
+/datum/element/light_eater/Attach(datum/target)
+	if(isatom(target))
+		if(ismovable(target))
+			RegisterSignal(target, COMSIG_MOVABLE_IMPACT, .proc/on_throw_impact)
+			if(isitem(target))
+				RegisterSignal(target, COMSIG_ITEM_AFTERATTACK, .proc/on_afterattack)
+				RegisterSignal(target, COMSIG_ITEM_HIT_REACT, .proc/on_hit_reaction)
+			else if(isprojectile(target))
+				RegisterSignal(target, COMSIG_PROJECTILE_ON_HIT, .proc/on_projectile_hit)
+	else if(istype(target, /datum/reagent))
+		RegisterSignal(target, COMSIG_REAGENT_EXPOSE_ATOM, .proc/on_expose_atom)
+	else
+		return ELEMENT_INCOMPATIBLE
+
+	return ..()
+
+/datum/element/light_eater/Detach(datum/source, force)
+	if(isatom(source))
+		if(ismovable(source))
+			UnregisterSignal(source, COMSIG_MOVABLE_IMPACT)
+			if(isitem(source))
+				UnregisterSignal(source, list(
+					COMSIG_ITEM_AFTERATTACK,
+					COMSIG_ITEM_HIT_REACT,
+				))
+			else if(isprojectile(source))
+				UnregisterSignal(source, COMSIG_PROJECTILE_ON_HIT)
+	else if(istype(source, /datum/reagent))
+		UnregisterSignal(source, COMSIG_REAGENT_EXPOSE_ATOM)
+	return ..()
+
+/**
+ * Makes the light eater consume all of the lights attached to the target atom.
+ *
+ * Arguments:
+ * - [food][/atom]: The atom to start the search for lights at.
+ * - [eater][/datum]: The light eater being used in this case.
+ */
+/datum/element/light_eater/proc/eat_lights(atom/food, datum/eater)
+	var/list/buffet = table_buffet(food)
+	if(!LAZYLEN(buffet))
+		return 0
+
+	. = 0
+	for(var/morsel in buffet)
+		. += devour(morsel, eater)
+
+	if(!.)
+		return
+
+	food.visible_message(
+		"<span class='danger'>Something dark in [eater] lashes out at [food] and [food.p_their()] light goes out in an instant!</span>",
+		"<span class='userdanger'>You feel something dark in [eater] lash out and gnaw through your light in an instant! It recedes just as fast, but you can feel that [eater.p_theyve()] left something hungry behind.</span>",
+		"<span class='danger'>You feel a gnawing pulse eat at your sight.</span>"
+	)
+
+/**
+ * Aggregates a list of the light sources attached to the target atom.
+ *
+ * Arguments:
+ * - [comissary][/atom]: The origin node of all of the light sources to search through.
+ * - [light_eater][/datum]: The light eater being applied to the target.
+ */
+/datum/element/light_eater/proc/table_buffet(atom/commisary, datum/light_eater)
```

Comment:
What is `datum/light_eater`? The component? If so, it shouldn't be passed around.
This doesn't seem to be used, though.

Reply:
It's not the component. I think I made it the parent to the light eater component in case whatever I'm hitting wants to react/change behavior according to that. It's a datum because I made the component compatible with reagents.

Reply:
> 
> 
> It's not the component. I think I made it the parent to the light eater component in case whatever I'm hitting wants to react/change behavior according to that. It's a datum because I made the component compatible with reagents.

Please rephrase, I didn't understand the answer.
What is `datum/light_eater` if not the component? Is this a reference to the element itself, the `src` of the proc? Or to something else?

Reply:
It is the reference to the datum the light eater element is attached to.